### ISSUE TITLE
Oauth refresh token

### ISSUE THREAD

**Issue by marijang** [mgonto/restangular#859]
I have 2 request in the "same time" so i need to refresh token only once. My example is getting 2x times for token. pls hlp

this is my code
```javascript
   refreshToken: function(){
        var deferred = $q.defer();
        var self = this;
        var token = Restangular.all('api/v1/oauth/access_token');
        var expired = function(token) {
            return (token && token.expires && new Date(token.expires) < new Date())
        };
        console.log('Called refresh token');
        if (expired(_token)){
            console.log('Token has expired and i m getting new one');
            var post_for = {
            'client_id': 1,
            'client_secret': 'iphone',
            'grant_type':'refresh_token',
            'refresh_token':_token.refresh_token,
            }
            token.post(post_for).then(function(data) {
              self.setToken(data);
              deferred.resolve(data);
            }, function(data){
              deferred.resolve(data);
            });
        }else{
          deferred.resolve(_token);
        }

        return deferred.promise;
      },
      service:function(){
         var self = this;
         var expired = function(token) {
            return (token && token.expires && new Date(token.expires) < new Date())
         };
         return Restangular.withConfig(function(RestangularConfigurer) {    
            if (_token){
              RestangularConfigurer.setDefaultHeaders({Authorization:'Bearer '+ _token.access_token});
              RestangularConfigurer.setBaseUrl('/eproject/public/api/v1/'); 
              RestangularConfigurer.setErrorInterceptor(function(response, deferred, responseHandler) {
                  if(response.status === 401) { // unauthorized
                      console.log('Token from settings is:'+_token.access_token);
                      if (angular.isDefined(_token)){
                       self.refreshToken().then(function(data) {
                          console.log('New token is set: '+_token.access_token);
                          response.config.headers.Authorization = 'Bearer '+ data.access_token;
                          $http(response.config).then(responseHandler, deferred.reject);
                          // Be aware that no request interceptors are called this way.
                        },function(){
                          alert('Error');
                        });
                      }
                      return false; // error handled
                    //}
                  }  

                  return true; // error not handled
              });

            }
          });  
      }
```

I m calling service for request. So when the user is unauthorized but have token i need it to refresh.
All my request are going for new token. How to prevent them?

**Comment by grabbou** [comment]
Please provide more readable, understandable explanation with code samples what's exactly going on (outgoing requests - which one are useless or which one have wrong parameters) and we will try to help. Refer to SO as well as this is a great place for problems like this one.
